book/src/secrets-and-credentials.md

```diff
@@ -80,11 +80,11 @@ The value of `csi.storage.k8s.io/controller-publish-secret-namespace` also suppo
 
 If specified, once provisioning is successful, the CSI `external-provisioner` sets the `CSIPersistentVolumeSource.ControllerPublishSecretRef` field in the new `PersistentVolume` object to refer to this secret.
 
-If specified, the CSI `external-attacher` attempts to fetch the secret referenced by the `CSIPersistentVolumeSource.ControllerPublishSecretRef` before an attach operation.
+If specified, the CSI `external-attacher` attempts to fetch the secret referenced by the `CSIPersistentVolumeSource.ControllerPublishSecretRef` before an attach or detach operation.
 
-If no such secret exists in the Kubernetes API, or the `external-attacher` is unable to fetch it, the attach operation fails.
+If no such secret exists in the Kubernetes API, or the `external-attacher` is unable to fetch it, the attach or detach operation fails.
 
-If the secret is retrieved successfully, the `external-attacher` passes it to the CSI driver in the `ControllerPublishVolumeRequest.secrets` field.
+If the secret is retrieved successfully, the `external-attacher` passes it to the CSI driver in the `ControllerPublishVolumeRequest.secrets` or `ControllerUnpublishVolumeRequest.secrets` field.
 
 ### Node Stage Secret
```

Comment:
and unstage?  also node unpublish further below

Reply:
nodeunstage and nodeunpublish do not have secrets.